pkg/apis/security/v1alpha1/httppolicy_types.go

```diff
@@ -0,0 +1,112 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// HTTPPolicy is the specification for HTTP traffic policy.
+type HTTPPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec HTTPPolicySpec `json:"spec"`
+}
+
+var (
+	_ apis.Validatable   = (*HTTPPolicy)(nil)
+	_ apis.Defaultable   = (*HTTPPolicy)(nil)
+	_ apis.HasSpec       = (*HTTPPolicy)(nil)
+	_ runtime.Object     = (*HTTPPolicy)(nil)
+	_ kmeta.OwnerRefable = (*HTTPPolicy)(nil)
+)
+
+// HTTPPolicySpec is the HTTPPolicy specification.
+type HTTPPolicySpec struct {
+	// JWT specifies the parameters to validate JTWs.
+	// If omitted, authentication will be skipped.
+	JWT *JWTSpec `json:"jwt,omitempty"`
+
+	// Rules is the list of rules to check for the policy.
+	// The rules should be evaludated in order. If the request under check
+	// passes one rule, it passes the policy check.
+	// If Rules is not specified, it implies the policy is to "allow all".
+	// If an empty rule is specified in Rules, it implies the policy is to "reject all".
+	Rules []HTTPPolicyRuleSpec `json:"rules,omitempty"`
+}
+
+// HTTPPolicyRuleSpec is the specification for a HTTP policy rule.
+// To pass a specified rule, a request must match all attributes provided in the rule.
+type HTTPPolicyRuleSpec struct {
+	// Auth is the authentication attribues to match.
+	Auth *RequestAuth `json:"auth,omitempty"`
+
+	// Operations is a list of operation attributes to match.
+	Operations []RequestOperation `json:"operations,omitempty"`
+
+	// Headers is a list of headers to match.
+	Headers []KeyValuesMatch `json:"headers,omitempty"`
+}
+
+// RequestAuth is the HTTP request authentication specification.
+type RequestAuth struct {
+	// Principals is a list of source identities to match.
+	// If omitted, it implies any principal is allowed.
+	Principals []string `json:"principals,omitempty"`
+
+	// Claims is a list of claims that should match certain patterns.
+	Claims []KeyValuesMatch `json:"claims,omitempty"`
+}
+
+// RequestOperation is the operation the request is taking.
+type RequestOperation struct {
+	// Hosts is a list of host names to match.
+	Hosts []StringMatch `json:"hosts,omitempty"`
+
+	// Paths is a list of paths to match.
+	Paths []StringMatch `json:"paths,omitempty"`
+
+	// Methods is a list of methods to match.
+	Methods []string `json:"methods,omitempty"`
```

Comment:
Would it be useful to allow things like:
for `/path1` methods are `POST`, `GET`... for `/path2` method `GET`, and so on...

Reply:
In general I don't think it matters much. You should always be able to construct multiple rules to achieve what you need. This API is currently a similar copy from https://istio.io/docs/reference/config/security/authorization-policy/#Operation.

Reply:
Might be worth adding a comment saying that this API has been influenced by Istio work or something, and add the link.. People will have more context... 
Again, optional..